### ISSUE TITLE
Prologue with shellcmd backend

### ISSUE THREAD

**Issue by HackerMD** [uzh/gc3pie#609]
I have problems related to execution of GC3Pie jobs within Docker containers.

Specifically, the execution of the command requires prior modifications of the `PATH`, but neither `.bashrc` nor `.bash_profile` files get sourced when bash is invoked in a non-interactive, non-login shell.

I found the `prologue_content` configuration setting, but it seems it's not supported for the `shellcmd` backend. I could create a custom entrypoint or startup script for containers to set the `PATH`, but it would be nice to be able to handle this on the side of GC3Pie. Do you see a way how this could be achieved?

**Comment by riccardomurri** [comment]
I agree the `prologue_*` and `epologue_*` settings should be honored by *all* backends, including `ShellcmdLrms`.

**Comment by riccardomurri** [comment]
As a workaround with the current code, just submit as task a shell-script that does whatever "prologue" actions you need before running the actual command.  For example::

```
my_docker_app = Application(
  ['my_docker.sh', 'run', 'ubuntu/xenial', 'lsb_release', '-a'],
  # ...
)
```

where the `my_docker.sh` script looks like the following::

```
#! /bin/sh

# prologue
echo "Setting up environment to run Docker ..."

# do it!
exec docker "$@"
```

**Comment by HackerMD** [comment]
The proposed workaround would break my application logic. I could solve it at the container level by ensuring the correct environment. Being able to use `prologue` with the `ShellcmdLrms` would certainly be useful in the long-term, however.
